ruma-api-macros/src/api/response.rs

```diff
@@ -290,52 +240,34 @@ impl ToTokens for Response {
             quote! { { #(#fields),* } }
         };
 
-        let (derive_deserialize, def) =
-            if let Some(body_field) = self.fields.iter().find(|f| f.is_newtype_body()) {
-                let field = Field { ident: None, colon_token: None, ..body_field.field().clone() };
+        let def = if let Some(body_field) = self.fields.iter().find(|f| f.is_newtype_body()) {
+            let field = Field { ident: None, colon_token: None, ..body_field.field().clone() };
 
-                // Though we don't track the difference between new type body and body
-                // for lifetimes, the outer check and the macro failing if it encounters
-                // an illegal combination of field attributes, is enough to guarantee
-                // `body_lifetimes` correctness.
-                let (derive_deserialize, lifetimes) = if self.has_body_lifetimes() {
-                    (TokenStream::new(), self.body_lifetimes())
-                } else {
-                    (quote!(::ruma_api::exports::serde::Deserialize), TokenStream::new())
-                };
-
-                (derive_deserialize, quote! { #lifetimes (#field); })
-            } else if self.has_body_fields() {
-                let fields = self.fields.iter().filter(|f| f.is_body());
-                let (derive_deserialize, lifetimes) = if self.has_body_lifetimes() {
-                    (TokenStream::new(), self.body_lifetimes())
-                } else {
-                    (quote!(::ruma_api::exports::serde::Deserialize), TokenStream::new())
-                };
+            quote! { (#field); }
+        } else if self.has_body_fields() {
+            let fields = self.fields.iter().filter(|f| f.is_body());
 
-                let fields = fields.map(ResponseField::field);
+            let fields = fields.map(ResponseField::field);
 
-                (derive_deserialize, quote!( #lifetimes { #(#fields),* }))
-            } else {
-                (TokenStream::new(), quote!({}))
-            };
+            quote!( { #(#fields),* } )
+        } else {
+            quote! { {} }
```

Comment:
While you're at it, maybe replace the upper `quote!(…)` with the more common `quote! { … }` too.